torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py

```diff
@@ -1299,13 +1299,19 @@ def _wait_for_post_backward(self) -> None:
     # A backward pass is done, clean up below.
     def _finalize_parameters(fsdp_module: XlaFullyShardedDataParallel) -> None:
       """Helper used below on all fsdp modules."""
+      frozen_params = []
```

Comment:
Can we identify which _free_full_params calls doesn't free the full params instead of adding a new pass at the end?

We might have bigger problem there.

Reply:
Will it be better to do it here?
https://github.com/pytorch/xla/blob/master/torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py#L1327

Reply:
After looking into the code more, I think this place is fine.